### ISSUE TITLE
Question: Where did you get the map definitions from?

### ISSUE THREAD

**Issue by cthrax** [kevalbhatt/timezone-picker#30]
Hi! Love what you have here, I'm in the middle of porting it to an angular library. I'm just curious where you came up with the map definitions for the polygons. Did you do this yourself? Or is there a base that you're linking up with moment-timezone? I'm just curious so I can update the definitions in the future.

Thanks again for the awesome project!

**Comment by kevalbhatt** [comment]
@cthrax Thanks.

Regarding your question:

I am using the JSON which is hardcoded. I am planning to make this JSON dynamic and customizable.

One more improvement, I am going to remove JQuery from the code.

if you are planning to create an Angular module you can use this plugin as I am going to remove the JQuery from the dependency.

Will commit the code in 4 - 5 days.

**Comment by cthrax** [comment]
@kevalbhatt thanks for the reply! Yeah, I saw that it was coming from the hardcoded json file. I was just wondering where the original data came from.

I spent some time investigating over the weekend and I think I have a solution for my needs anyway to keep the timezone information up to date. [This tool](https://github.com/evansiroky/timezone-boundary-builder/releases) scrapes data from open maps to get timezone boundary information and makes a release of geojson with that info. It's a very detailed set of coordinates though, so I ran through another tool, [MapShaper](https://mapshaper.org/) to reduce the size of the geojson. Finally I'm combining that output with moment-timezone's list of timezone data to create a timezone.json similar to what you have right now. It's still quite a bit bigger than your original json, so I'm still playing with that.

Anyway, nice work! I look forward to seeing what you come up with for the vanilla version. My angular port was really simple thanks to what you had in place. I was able to extract out all the DOM manipulation into angular templates and got all my bindings via angular, so it was an extremely low amount of code on my end. I got hung up making it generic and keeping some of your features, like quicklinks and the dropdown, so I'm in the process of making a couple dependent angular libraries before I release it wild. That way someone can choose to have their own select library, or not. Rather than forcing my select library on them.
